vpr/src/route/connection_box_lookahead_map.cpp

```diff
@@ -153,57 +120,39 @@ int CostMap::node_to_segment(int from_node_ind) const {
     return segment_map_[from_node_ind];
 }
 
+static util::Cost_Entry penalize(const util::Cost_Entry& entry, int distance, float penalty) {
+    penalty = std::max(penalty, PENALTY_MIN);
+    return util::Cost_Entry(entry.delay + distance * penalty * PENALTY_FACTOR,
+                            entry.congestion);
+}
+
 // get a cost entry for a segment type, connection box type, and offset
-util::Cost_Entry CostMap::find_cost(int from_seg_index, ConnectionBoxId box_id, int delta_x, int delta_y, int* out_of_bounds_val) const {
+util::Cost_Entry CostMap::find_cost(int from_seg_index, ConnectionBoxId box_id, int delta_x, int delta_y) const {
     VTR_ASSERT(from_seg_index >= 0 && from_seg_index < (ssize_t)offset_.size());
     const auto& cost_map = cost_map_[from_seg_index][size_t(box_id)];
     if (cost_map.dim_size(0) == 0 || cost_map.dim_size(1) == 0) {
         return util::Cost_Entry();
     }
 
-    int dx = delta_x - offset_[from_seg_index][size_t(box_id)].first;
-    int dy = delta_y - offset_[from_seg_index][size_t(box_id)].second;
-
-    int out_of_bounds_distance = 0;
-    if (dx < 0) {
-        out_of_bounds_distance += abs(dx);
-        dx = 0;
-    }
-    if (dy < 0) {
-        out_of_bounds_distance += abs(dy);
-        dy = 0;
-    }
-
-    if (dx >= (ssize_t)cost_map.dim_size(0)) {
-        out_of_bounds_distance += abs(int(dx) - int(cost_map.dim_size(0) - 1));
-        dx = cost_map.dim_size(0) - 1;
-    }
-    if (dy >= (ssize_t)cost_map.dim_size(1)) {
-        out_of_bounds_distance += abs(int(dy) - int(cost_map.dim_size(1) - 1));
-        dy = cost_map.dim_size(1) - 1;
-    }
-
-    if (out_of_bounds_val != nullptr) {
-        *out_of_bounds_val = out_of_bounds_distance;
-    }
-
-    return cost_map_[from_seg_index][size_t(box_id)][dx][dy];
-}
-
-static util::Cost_Entry penalize(const util::Cost_Entry& entry, int distance, float penalty) {
-    auto out = util::Cost_Entry(entry.delay + distance * penalty * PENALTY_FACTOR,
-                                entry.congestion);
-
-    out.fill = entry.fill;
-    return out;
+    vtr::Point<int> coord(delta_x - offset_[from_seg_index][size_t(box_id)].first,
+                          delta_y - offset_[from_seg_index][size_t(box_id)].second);
+    vtr::Rect<int> bounds(0, 0, cost_map.dim_size(0), cost_map.dim_size(1));
+    auto closest = closest_point_in_rect(bounds, coord);
+    auto cost = cost_map_[from_seg_index][size_t(box_id)][closest.x()][closest.y()];
```

Comment:
Won't this explode on empty cost_map's?  The previous code had logic to detect that, this code no longer does?

Reply:
It's checked a few lines up:
```c
    if (cost_map.dim_size(0) == 0 || cost_map.dim_size(1) == 0) {
        return util::Cost_Entry();
    }
```